app/src/main/java/com/example/sw0b_001/Homepage/GatewayClientListingFragment.kt

```diff
@@ -130,7 +148,8 @@ class GatewayClientListingFragment : Fragment(R.layout.activity_gateway_clients_
     }
 
     class GatewayClientListingAdapter(private val gatewayClientsCommunications: GatewayClientsCommunications,
```

Comment:
@Idadelveloper You shouldn't pass viewModel to the Adapter, they always have to be independent of each other